app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -751,6 +759,23 @@ class HomeFragment : Fragment() {
         lifecycleScope.launch(IO) {
             requireComponents.reviewPromptController.promptReview(requireActivity())
         }
+
+        if (FeatureFlags.showWallpapers) {
+            binding.wordmark.setOnClickListener {
+                val settings = context.settings()
+                val newWallpaper = settings.getWallpaper().switchToNextWallpaper(settings)
+                updateWallpaper(newWallpaper)
+            }
+        }
+    }
+
+    private fun updateWallpaper(newWallpaper: Wallpaper) {
+        val context = requireContext()
+        if (newWallpaper == Wallpaper.NONE) {
+            binding.homeLayout.setBackgroundColor(context.getColorFromAttr(newWallpaper.drawable))
+        } else {
+            binding.homeLayout.setBackgroundResource(newWallpaper.drawable)
```

Comment:
when the wallpaper is set here, how does the image get rendered on the device? is it adjusted to the aspect ratio of the device or is it centered within the screen?

Reply:
It should be adjusted to the aspect ratio of the device. We're still waiting on finalized designs to do this testing, but we're going to (hopefully) try this out with both a vector graphic and PNGs. We should have portrait and landscape versions of both. The vector will autoscale, and the PNGs would match based on [the screen density of the device](https://developer.android.com/training/multiscreen/screendensities)